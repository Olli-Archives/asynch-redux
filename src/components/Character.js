import React from 'react';
import PropTypes from 'prop-types';
export default function Character({ character }){


  return (
    <section>
      <img src={character.photoUrl}/>
      <p> {character.name}</p>
    </section>
  );
}

Character.propTypes = {
  character:PropTypes.object.isRequired
}
;
