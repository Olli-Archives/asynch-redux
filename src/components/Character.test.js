import React from 'react';
import Character from './Character';
import { shallow } from 'enzyme';

describe('Character component', ()=>{

  const testObject = {
    image:'test image',
    name:'test name'
  };

  it('renders a character', ()=>{
    const wrapper = shallow(<Character image={testObject.image} name={testObject.name}/>);
    expect(wrapper).toMatchSnapshot();
  });
})
;
