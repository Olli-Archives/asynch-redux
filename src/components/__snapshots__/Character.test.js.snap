// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`character component renders a character 1`] = `
<section>
  <img
    src="test url"
  />
  <p>
     
    test name
  </p>
</section>
`;
